components/client/src/main/java/com/hotels/styx/client/loadbalancing/strategies/PowerOfTwoStrategy.java

```diff
@@ -61,17 +63,18 @@ public LoadBalancer create(Environment environment, Configuration strategyConfig
 
     @Override
     public Optional<RemoteHost> choose(LoadBalancer.Preferences preferences) {
-        RemoteHost[] hosts = toArray(activeOrigins.snapshot(), RemoteHost.class);
+        List<RemoteHost> hosts = StreamSupport.stream(activeOrigins.snapshot().spliterator(), false)
+                .collect(Collectors.toList());
```

Comment:
This is a data path code, and may get invoked once per request.

Do you know how the proposed alternative would work performance wise?

Reply:
Hm. In non-test scenarios, the main (only?) implementation of `activeOrigins.snapshot()` is in `OriginsInventory`, and the Iterable is created in `OriginsInventory.pools(x)`, and is actually already a `List`. So yes, this can be more performant (i.e. use a List if we already have one, and convert more efficiently to a List otherwise)